seabios: 1.16.2 -> 1.16.3 (plus misc updates)

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Seabios:

- finalAttrs design pattern
- split outputs
- generate config file instead of inlining it
- tag distribution in EXTRAVERSION
- install the Csm16.bin biosfile to $out/share/seabios/ instead of $out/
- remove nested with
- add myself as maintainer

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
seabios: 1.16.2 -> 1.16.3

AndersonTorres committed:
seabios: migrate to by-name

AndersonTorres committed:
xen_4_15: fixup seabios location

Since it is now on `${seabios}/share/seabios`.

AndersonTorres committed:
ovmf: fixup seabios location

Since it is now on `${seabios}/share/seabios`.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3012

adamcstephens commented:
The title indicates this is a simple bump, but the changes don't align with my expectations from such a version change. Can you provide more information into why the other changes are necessary?

Is there a way to test seabios? Does OVMF use seabios for CSM enabled firmwares?

AndersonTorres commented:
> The title indicates this is a simple bump, but the changes don't align with my expectations from such a version change. Can you provide more information into why the other changes are necessary?

I  have split the changes in "self-contained" units.
- The version bump was picked from ryanbot
- The finalAttrs design pattern and split outputs are standard
- The biosfile was moved from the "root" directory to a corresponding "LFS" directory entry.
- It implies the older references to such biosfile are now broken.

> 
> Is there a way to test seabios? 

Qemu is the most straightforward method.

> Does OVMF use seabios for CSM enabled firmwares?

Yes, according to the current package.

RaitoBezarius commented on pkgs/by-name/se/seabios/package.nix:
```diff
@@ -0,0 +1,57 @@
+    license = lib.licenses.lgpl3;
+    maintainers = with lib.maintainers; [ AndersonTorres ];
+    platforms = with lib.platforms;
+      lib.intersectLists (unix) (x86);
```
I think we use inspect patterns rather than intersection for this type of stuff.

AndersonTorres replied:
I don't know about them. I have seen them once while reviewing some code from Alyssais, but I didn't understand.

RaitoBezarius commented on pkgs/by-name/se/seabios/package.nix:
```diff
@@ -0,0 +1,57 @@
+  hardeningDisable = [ "pic" "stackprotector" "fortify" ];
+
+  postConfigure = ''
+    cp ${./config.txt} .config
```
Why must the file be separate rather inlined?

AndersonTorres replied:
Not a "must" but a preference. I believe it is cleaner.
The other possibility is to generate it via some converter like `toINI`.

RaitoBezarius commented:
Unfortunately, we don't have a great test suite for SeaBIOS stuff, I would invite you to take ownership of the BIOS tests we have in `nixos/tests/boot.nix` if possible.

AndersonTorres committed:
seabios: refactor

- finalAttrs design pattern
- split outputs
- generate config file instead of inlining it
- tag distribution in EXTRAVERSION
- install the Csm16.bin biosfile to $out/share/seabios/ instead of $out/
- remove nested with
- add myself as maintainer

AndersonTorres committed:
seabios: mark Darwin as a bad platform

Maybe it is broken on Clang...

AndersonTorres commented on pkgs/by-name/se/seabios/package.nix:
```diff
@@ -0,0 +1,72 @@
+  '';
+
+  meta = {
+    homepage = "http://www.seabios.org";
```
Done

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3093

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1326
